### PR TITLE
feat: Add Credentials API

### PR DESCRIPTION
**Pull request by mykola-vrmchk**

### CODE REVIEW

**Comment by mykola-vrmchk** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2502** <a href="https://app.graphite.dev/github/pr/dxos/dxos/2502" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dxos/dxos/2502?utm_source=stack-comment).

File: packages/core/halo/credentials/src/presentations/verifier.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+  const proofVerification = await Promise.all(
+    presentation.proofs?.map((proof) => {
+      if (!proof.chain) {
+        return {
+          kind: 'fail',
+          errors: ['Delegated credential is missing credential chain.']
+        };
+      }
+      // TODO(mykola): Add chain verification?
```
Comment (dmaretskyi) [inline packages/core/halo/credentials/src/presentations/verifier.ts:30]: you could use the same code as in credential verification

File: packages/core/protocols/src/proto/dxos/client.proto [hunk 2]
```diff
@@ -116,10 +119,17 @@ message Contacts {
   repeated Contact contacts = 1;
 }
 
+message SignPresentationRequest {
+  dxos.halo.credentials.Presentation presentation = 1;
+  bytes nonce = 2;
```
Comment (dmaretskyi) [inline packages/core/protocols/src/proto/dxos/client.proto:124]: optional

**Review by dmaretskyi — APPROVED** [review]

File: packages/core/halo/credentials/src/presentations/verifier.ts (hunk 1 above)
Comment (mykola-vrmchk) [reply packages/core/halo/credentials/src/presentations/verifier.ts:30]: I will add it in a separate PR

**Pull request merged**